File: src/main/java/life/magpies/registry/commands/CommandTpcAetp.java
package life.magpies.registry.commands;

import com.mojang.brigadier.arguments.StringArgumentType;
import com.mojang.brigadier.context.CommandContext;
import life.magpies.registry.util.ModManager;
import life.magpies.registry.util.TeleportRequest;
import life.magpies.registry.util.TeleportRequestManager;
import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback;
import net.minecraft.server.command.CommandManager;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.text.Text;
import net.minecraft.util.Formatting;

import java.util.UUID;

import static net.minecraft.server.command.CommandManager.literal;

public class CommandTpcAetp {
    public static void register() {
        CommandRegistrationCallback.EVENT.register((dispatcher, registryAccess, environment) -> {
            dispatcher.register(literal("tpcept")
                    .requires(source -> source.hasPermissionLevel(0))
                    .then(CommandManager.argument("player", StringArgumentType.word())
                            .suggests(PlayerSuggestionProvider.suggestPlayers)
                            .executes(ctx -> {
                                String targetPlayer = StringArgumentType.getString(ctx, "player");
                                return execute(ctx, targetPlayer);
                            })
                    ));
        });
    }

    public static int execute(CommandContext<ServerCommandSource> ctx, String target) {
        //获取命令源
        ServerCommandSource source = ctx.getSource();
        ServerPlayerEntity player = source.getPlayer();
        // 获取uuid
        if (source.isExecutedByPlayer()) {
            UUID SourceUUID = source.getPlayer().getUuid();
            ServerPlayerEntity Target = source.getServer().getPlayerManager().getPlayer(target);
            UUID TargetUUID = Target.getUuid();

            TeleportRequestManager manager = ModManager.manager;
            TeleportRequest request = manager.getRequest(TargetUUID,SourceUUID);

            if (request != null && request.isValid()) {
                source.sendFeedback(() -> Text.literal("您已接授请求").formatted(Formatting.GREEN), false);
                Target.sendMessage(Text.literal("对方已接授请求").formatted(Formatting.GREEN), false);
                manager.remove(request);
                player.teleport(Target.getServerWorld(), Target.getPos().x, Target.getPos().y, Target.getPos().z, Target.getYaw(), Target.getPitch());
            } else {
                source.sendFeedback(() -> Text.literal("传送请求失效了").formatted(Formatting.RED), false);
            }
        }
        return 1;
    }
}

File: src/main/kotlin/life/magpies/registry/commands/CommandTpcAetp.kt
package life.magpies.registry.commands

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.arguments.StringArgumentType
import com.mojang.brigadier.context.CommandContext
import life.magpies.utils.ModManager
import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback
import net.minecraft.command.CommandRegistryAccess
import net.minecraft.server.command.CommandManager
import net.minecraft.server.command.CommandManager.RegistrationEnvironment
import net.minecraft.server.command.ServerCommandSource
import net.minecraft.text.Text
import net.minecraft.util.Formatting

object CommandTpcAetp {
    fun register() {
        CommandRegistrationCallback.EVENT.register(CommandRegistrationCallback { dispatcher: CommandDispatcher<ServerCommandSource?>, registryAccess: CommandRegistryAccess?, environment: RegistrationEnvironment? ->
            dispatcher.register(
                CommandManager.literal("tpcept")
                    .requires { source: ServerCommandSource ->
                        source.hasPermissionLevel(
                            0
                        )
                    }
                    .then(
                        CommandManager.argument("player", StringArgumentType.word())
                            .suggests(PlayerSuggestionProvider.suggestPlayers)
                            .executes { ctx: CommandContext<ServerCommandSource> ->
                                val targetPlayer = StringArgumentType.getString(ctx, "player")
                                execute(ctx, targetPlayer)
                            }
                    )
            )
        })
    }

    private fun execute(ctx: CommandContext<ServerCommandSource>, targetPlayer: String?): Int {
        //获取命令源
        val source = ctx.source
        val player = CommandUtils.getPlayer(ctx) ?: return 0
        // 获取uuid
        if (source.isExecutedByPlayer) {
            val sourceUUID = player.uuid
            val target = source.server.playerManager.getPlayer(targetPlayer)
            val targetUUID = target!!.uuid

            val manager = ModManager.manager
            val request = manager.getRequest(targetUUID, sourceUUID)

            if (request != null && request.isValid) {
                source.sendFeedback({
                    Text.literal(
                        "您已接授请求"
                    ).formatted(Formatting.GREEN)
                }, false)
                target.sendMessage(Text.literal("对方已接授请求").formatted(Formatting.GREEN), false)
                manager.remove(request)
                player.teleport(
                    target.serverWorld,
                    target.pos.x,
                    target.pos.y,
                    target.pos.z,
                    target.yaw,
                    target.pitch
                )
            } else {
                source.sendFeedback({
                    Text.literal(
                        "传送请求失效 获者您未有请求"
                    ).formatted(Formatting.RED)
                }, false)
            }
        }
        return 1
    }
}